runtimebp/internal/maxprocs/maxprocs.go

```diff
@@ -0,0 +1,156 @@
+// Package maxprocs optimizes GOMAXPROCS for Infrared environments.
+// It attempts to balance application performance with cluster-level resource utilization.
+// Applications may tune GOMAXPROCS as necessary (see Set).
+// The defaults in this package are subject to change.
+//
+// NOTE: this is manually copied from internal baseplate/v2/internal/maxprocs/maxprocs.go
+package maxprocs
+
+import (
+	"fmt"
+	"math"
+	"os"
+	"runtime"
+	"strconv"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+	ubermaxprocs "go.uber.org/automaxprocs/maxprocs"
+)
+
+type floatEnv struct {
+	key   string
+	gauge *prometheus.GaugeVec
+
+	present bool
+	raw     string
+	val     float64
+}
+
+const (
+	// The default scale applies when request is set but not scale (or scale is
+	// invalid). It defaults to some "oversubscription" to take advantage of
+	// excess Kubernetes node capacity when available.
+	//
+	// Extremely performance-sensitive services should set scale to 1.0 to
+	// ensure consistent performance.
+	//
+	// This value was NOT chosen scientifically and IS subject to change.
+	defaultCPURequestScale = 1.5
+
+	setByGOMAXPROCS   = "gomaxprocs"
+	setByRequest      = "cpu_request"
+	setByAutomaxprocs = "automaxprocs"
+)
+
+var (
+	mEnvGOMAXPROCS = promauto.NewGaugeVec(prometheus.GaugeOpts{
+		Name: "baseplate_go_env_gomaxprocs",
+		Help: "Value of the GOMAXPROCS environment variable at startup. 0 if not a number",
+	}, []string{"status"})
+	mEnvCPURequest = promauto.NewGaugeVec(prometheus.GaugeOpts{
+		Name: "baseplate_go_env_baseplate_cpu_request",
+		Help: "Value of the BASEPLATE_CPU_REQUEST environment variable at startup. 0 if not a number",
+	}, []string{"status"})
+	mEnvCPURequestScale = promauto.NewGaugeVec(prometheus.GaugeOpts{
+		Name: "baseplate_go_env_baseplate_cpu_request_scale",
+		Help: "Value of the BASEPLATE_CPU_REQUEST_SCALE environment variable at startup. 0 if not a number",
+	}, []string{"status"})
```

Comment:
🔕 I'm wondering if it makes more sense to combine this to a single gauge with an additional env_var label. depending on how likely/unlikely we are going to handle more env vars or change env var names in the future.